app/views/govuk_publishing_components/components/_layout_header.html.erb

```diff
@@ -13,63 +15,49 @@
-            <%= environment %>
-          </span>
-        <% end %>
-      </a>
-    </div><div class="govuk-header__content gem-c-header__content">
-    <% if navigation_items.any? %>
-    <button role="button" class="govuk-header__menu-button gem-c-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
-    <nav class="gem-c-header__nav">
-      <ul id="navigation" class="govuk-header__navigation govuk-header__navigation--end" aria-label="Top Level Navigation">
-        <% navigation_items.each_with_index do |item, index| %>
-          <li class="govuk-header__navigation-item <%= "govuk-header__navigation-item--active" if item[:active] %>
-            <%= "govuk-header__navigation-item--collapsed-menu-only" if item[:show_only_in_collapsed_menu] %>">
-            <%= link_to(
-              item[:text],
-              item[:href],
-              class: 'govuk-header__link',
-              ) %>
-          </li>
-        <% end %>
-      </ul>
-    </nav>
-    <% end %>
-
-    </div>
-    </div>
-
-    <% if search %>
-        <div class="govuk-grid-column-one-third gem-c-layout-header__search">
+<% if search_left %>
+  <header class="gem-c-layout-header gem-c-layout-header--search-left govuk-header gem-c-layout-header--<%= environment %> <%= "gem-c-layout-header--no-bottom-border" if remove_bottom_border %>" role="banner" data-module="govuk-header">
+    <div class="govuk-header__container <%= width_class %>">
+      <div class="govuk-grid-row govuk-!-margin-left-0 govuk-!-margin-right-0">
+        <div class="gem-c-layout-header__logo govuk-grid-column-one-third-from-desktop">
+            <%= render "govuk_publishing_components/components/layout_header/header_logo", public_environment: public_environment, environment: environment, product_name: product_name %>
+        </div>
+      </div>
+      <div  class="govuk-grid-row govuk-!-margin-left-0 govuk-!-margin-right-0">
+        <div class="govuk-grid-column-full govuk-grid-column-one-third-from-desktop gem-c-layout-header__search">
           <%= render "govuk_publishing_components/components/search", {
             id: "site-search-text",
             label_text: "Search",
             no_border: true,
             no_margin: true
           } %>
         </div>
-    <% end %>
+        <div class="govuk-header__content gem-c-header__content govuk-grid-column-full govuk-grid-column-two-thirds-from-desktop">
+          <%= render "govuk_publishing_components/components/layout_header/navigation_items", navigation_items: navigation_items  %>
+        </div>
+      </div>
+    </div>
+  </header>
 
-  </div>
-</header>
+<% else %>
+  <header class="gem-c-layout-header govuk-header gem-c-layout-header--<%= environment %> <%= "gem-c-layout-header--no-bottom-border" if remove_bottom_border %>" role="banner" data-module="govuk-header">
```

Comment:
This bit could also be clearer with this approach: https://github.com/alphagov/govuk_publishing_components/blob/master/app/views/govuk_publishing_components/components/_textarea.html.erb#L23-L25